### PR TITLE
EntityCardWidget: Fixed deletion of list items while iterating.

### PR DESCRIPTION
**Pull request by eliveikis**
The `remove_field` function was deleting the element from the `_fields` list and it was being called from iterations of that list.

This change separates the destruction of the field and its deletion from the list.

### CODE REVIEW

**Comment by eliveikis** [comment]
@manneohrstrom Please review. There were other changes made in September that also combined to fix https://jira.autodesk.com/browse/GAME-17417, this is that last remaining piece.

File: python/version_details/shotgun_entities/card_widget.py [hunk 1]
```diff
@@ -186,20 +187,27 @@ def remove_field(self, field_name):
...
+        field_widget.deleteLater()
 
         # If there's a label, then also remove that.
         field_label = self._fields[field_name]["label"]
 
         if field_label:
             field_label.hide()
             self.ui.field_grid_layout.removeWidget(field_label)
+            field_label.setParent(None)
+            field_label.deleteLater()
```
Comment (eliveikis) [inline python/version_details/shotgun_entities/card_widget.py]: @manneohrstrom, @thebeeland   Is this necessary? I saw this pattern used elsewhere in toolkit when removing widgets and we want to make sure things are cleaned up properly so I added this here and above.

**Comment by eliveikis** [comment]
will push another version of this soon

**Comment by gomezll** [comment]
@thebeeland , @manneohrstrom Are you guys OK with these changes? If so, approve for merge. Thanks!

File: python/version_details/shotgun_entities/card_widget.py (hunk 1 above)
Comment (thebeeland) [inline python/version_details/shotgun_entities/card_widget.py]: I suspect this pattern is to plug a memory leak. I don't have first-hand experience with it, but I think it's reasonable to use it here given the situation.

**Pull request merged**